### ISSUE TITLE
error handling

### ISSUE THREAD

**Issue by astropanic** [levigross/grequests#4]
Is there a reason why this library doesn't provide the "go way" of error handling ? 

```
resp := grequests.Get("http://httpbin.org/xml", nil)

if resp.Error != nil {
    log.Fatalln("Unable to make request", resp.Error)
}
```

VS

```
resp, err := grequests.Get("http://httpbin.org/xml", nil)

if err != nil {
    log.Fatalln("Unable to make request", err)
}
```

**Comment by ShawnMilo** [comment]
The standard library uses  both methods. The second is more common, but `Scanner.Scan()

**Comment by levigross** [comment]
The `resp.Error` pattern is a remnant of the asynchronous nature of the library (up until a few days ago all functions returned a channel). 

I do intend to make to change the API to the more "standard" way

```go
resp, err := grequests.Get("http://httpbin.org/xml", nil)

if err != nil {
    log.Fatalln("Unable to make request", err)
}
```

**Comment by levigross** [comment]
Fixed in #0.2
